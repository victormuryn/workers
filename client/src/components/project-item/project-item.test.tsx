import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';

import ProjectItem from './project-item';

describe(`ProjectItem tests`, () => {
  it(`ProjectItem renders corrects`, () => {
    const tree = renderer.create(<Router>
      <ProjectItem
        _id="111"
        title="Title"
        price={123}
        date={`Wed Feb 17 2021 19:52:02 GMT+0200
        (Eastern European Standard Time)`}
      />
    </Router>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
