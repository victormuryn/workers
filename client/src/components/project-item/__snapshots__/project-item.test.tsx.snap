// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem tests ProjectItem renders corrects 1`] = `
<li
  className="projects__item"
>
  <div
    className="align-items-center row"
  >
    <div
      className="col-md-7 offset-md-1"
    >
      <a
        className="project__item-link"
        href="/project/111"
        onClick={[Function]}
      >
        <span
          className="h4"
        >
          Title
        </span>
        <p
          className="small text-muted mt-2 mb-md-0"
        >
          Програмування
        </p>
      </a>
    </div>
    <div
      className="col-md-1"
    >
      <p
        className="project__item-text text-success fw-bold m-md-0"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        123 ₴
      </p>
    </div>
    <div
      className="col-md-1"
    >
      <p
        className="project__item-text text-danger m-md-0"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        15
      </p>
    </div>
    <div
      className="col-md-1"
    >
      <p
        className="project__item-text m-md-0"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        17 лютого
      </p>
    </div>
  </div>
</li>
`;
